### DISCUSSION TITLE
How to use Stryd for Pace and Treadmill for Incline

### DISCUSSION THREAD

**Discussion by afitz85** [cagnulein/qdomyos-zwift#2010] [Q&A]
Hello,
how can i use Stryd for pace and my treadmill (Horizon Paragon X) for Incline?
Thank you.
Andreas Fitz

**Comment by cagnulein** [comment]
yes! just add the stryd under accessories, power sensor. let me know if you
have any issues

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno ven 19 gen 2024 alle 08:26 afitz85 ***@***.***> ha
scritto:

**Reply by afitz85** [reply]
ok, but it connects the treadmill automatically to record the incline?
even if stryd is connected as power sensor, zwift takes the speed and not the wattage?
Thanks.

**Reply by cagnulein** [reply]
yes it will connect automatically to the treadmill and also then to the sensor.
it will take by default speed and inclination from the treadmill and power from the sensor.
If you want to use the pace from the stryd, I have to do a modification but it's also possible. Could I ask you why you would like to take the pace from the stryd? it should be less real than the speed from the treadmill itself

**Reply by afitz85** [reply]
but the power from the stryd is just saved to the activity file, it's irrelevant and not used for speed in zwift?
for me the stryd is more accurate at speed than my treadmill... i did a little bit of testing here and i can say for me it's very accurate when i look at power/heartrate/pace, also compared to oudside.
my treadmill shows about 0,7 km/h faster then real pace (stryd).
i would say it's very rare that a treadmill really matches the real speed, so the possibility to use the pace instead of power from the stryd should be interesting for more people than just me :)

But as i see i can also offset my treadmill speed 0,7 km/h so that it matches the stryd, am i right?

**Reply by cagnulein** [reply]
> but the power from the stryd is just saved to the activity file, it's irrelevant and not used for speed in zwift? 

yes, zwift uses only the speed from qz.

> for me the stryd is more accurate at speed than my treadmill... i did a little bit of testing here and i can say for me it's very accurate when i look at power/heartrate/pace, also compared to oudside.

strange, did you think about getting a treadmill calibrator (quite cheap on amazon) and adjust it using the speed gain factor on qz?

> my treadmill shows about 0,7 km/h faster then real pace (stryd). i would say it's very rare that a treadmill really matches the real speed, so the possibility to use the pace instead of power from the stryd should be interesting for more people than just me :)

sure i can do it, no problem but try to use the speed gain or the speed offset setting in qz, maybe you just don't need any change. if it's just an offset, put a -0.7 in the speed offset and you're good to go already.

> 
> But as i see i can also offset my treadmill speed 0,7 km/h so that it matches the stryd, am i right?

exactly :)

**Reply by afitz85** [reply]
ok thanks, i will try that things!

never heard of a "treadmill calibrator", may you have a link for me?
there is unfortunately no build in calibration on my treadmill

**Comment by cagnulein** [comment]
this one
https://www.amazon.it/gp/product/B00MNEC9DW/ref=ox_sc_saved_image_1?smid=A3UMU276GPXXPH&psc=1

very cheap :)

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno ven 19 gen 2024 alle ore 10:22 afitz85 ***@***.***>
ha scritto:

**Reply by afitz85** [reply]
i don't know what i'm doing wrong. i just tried it with the android app and i can't make it run... after several tries i connected to the treadmill and i saw my tiles, but the speed or incline of my treadmill just doesn't show up, it stays at "0"

then i tried it with the pc version (what suits me better because of zwift on the same pc), and i get the connection, no problem so far but the values of speed and incline just also stays at "0", even when i press "start" the elapsed time is not running.

i just made it once that the speed and incline shows up (don't know why), but with the next start of the program it doesn't work anymore (stays at "0" all the time).

i don't know what i'm doing wrong... any suggestions here for me? :(

**Comment by cagnulein** [comment]
collect a debug log and i will check it
https://github.com/cagnulein/qdomyos-zwift/wiki/How-do-i-get-the-debug-log-in-case-something-doesn't-work%3F


Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno ven 19 gen 2024 alle 13:37 afitz85 ***@***.***> ha
scritto:
